clang/lib/CIR/CodeGen/CIRGenFunction.h

```diff
@@ -1188,6 +1190,16 @@ class CIRGenFunction : public CIRGenTypeCache {
   void buildNullabilityCheck(LValue LHS, mlir::Value RHS,
                              clang::SourceLocation Loc);
 
+  /// Same as IRBuilder::CreateInBoundsGEP, but additionally emits a check to
```

Comment:
Maybe it should be part of `CIRGenBuilder.h` then?

Reply:
I don't think so. The `llvm_unreacheable` hides a substantial amount of functionality from the original codegen path. I'll complement the `llvm_unreacheable` to make this clear.